sys/kern/kern_descrip.c

```diff
@@ -923,7 +924,20 @@ kern_fcntl(struct thread *td, int fd, int cmd, intptr_t arg)
 		break;
 
 	default:
-		error = EINVAL;
+		if ((cmd & ((1u << F_DUP3FD_SHIFT) - 1)) != F_DUP3FD) {
+			error = EINVAL;
```

Comment:
Consider using EXTERROR() to provide more context to userspace about EINVAL cause.